3rdparty/fastcv/src/fastcv_hal_imgproc.cpp

```diff
@@ -743,4 +743,253 @@ int fastcv_hal_warpPerspective(
+        dstHeight_ = (height_ + 2*n + 1)/2;
+        dstWidth_ = (width_ + 2*n + 1)/2;
+        origDstHeight_ = (height_ + 1)/2;
+        origDstWidth_ = (width_ + 1)/2;
+
+        cv::Mat dst_padded = cv::Mat(dstHeight_, dstWidth_, CV_8U);
+
+        fcvPyramidLevel_v2 *framePyr = new fcvPyramidLevel_v2[2];
+        framePyr[0].ptr = NULL;
+        framePyr[1].ptr = dst_padded.data;
+        framePyr[1].stride = dstWidth_;
+
+        fcvPyramidCreateu8_v4(src_.data, width_ + 2*n, height_ + 2*n,
+                                 width_ + 2*n, 2, FASTCV_PYRAMID_SCALE_HALF,
+                                 framePyr, FASTCV_BORDER_UNDEFINED, 0);
+
+        int start_val = stripeHeight * range.start;
+        cv::Mat dst_temp1 = dst_padded(cv::Rect(n/2, n/2, origDstWidth_, origDstHeight_));
+        cv::Mat dst_temp2 = dst(cv::Rect(0, start_val/2, origDstWidth_, origDstHeight_));
+        dst_temp1.copyTo(dst_temp2);
+    }
+
+private:
+    cv::Mat src;
+    const int width;
+    const int height;
+    cv::Mat dst;
+    const int bdr;
+    const int knl;
+    const int stripeHeight;
+    const int nStripes;
+
+    FcvPyrLoop_Invoker(const FcvPyrLoop_Invoker &);  // = delete;
+    const FcvPyrLoop_Invoker& operator= (const FcvPyrLoop_Invoker &);  // = delete;
+};
+
+int fastcv_hal_pyrdown(
+    const uchar*     src_data,
+    size_t           src_step,
+    int              src_width,
+    int              src_height,
+    uchar*           dst_data,
+    size_t           dst_step,
+    int              dst_width,
+    int              dst_height,
+    int              depth,
+    int              cn,
+    int              border_type)
+{
+    if(depth != CV_8U || cn!= 1)
+    {
+        CV_HAL_RETURN_NOT_IMPLEMENTED("src type not supported");
+    }
+
+    int dstW = (src_width & 1)  == 1 ? ((src_width + 1)  >> 1) : ((src_width) >> 1);
+    int dstH = (src_height & 1) == 1 ? ((src_height + 1) >> 1) : ((src_height) >> 1);
+
+    if((dstW > dst_width) || (dstH > dst_height))
+    {
+        CV_HAL_RETURN_NOT_IMPLEMENTED("dst size needs to be atleast half of the src size");
+    }
+
+    INITIALIZATION_CHECK;
+
+    fcvBorderType bdr;
+    uint8_t bVal = 0;
+    int nThreads = cv::getNumThreads();
+    if(nThreads <= 1)
+    {
+        switch(border_type)
+        {
+            case cv::BORDER_REPLICATE:
+                bdr = FASTCV_BORDER_REPLICATE;
+                break;
+            case cv::BORDER_REFLECT:
+                bdr = FASTCV_BORDER_REFLECT;
+                break;
+            case cv::BORDER_REFLECT101:    // cv::BORDER_REFLECT_101, BORDER_DEFAULT
+                bdr = FASTCV_BORDER_REFLECT_V2;
+                break;
+            default:
+                CV_HAL_RETURN_NOT_IMPLEMENTED("border type not supported");
+        }
+
+        fcvPyramidLevel_v2 *frame1Pyr = new fcvPyramidLevel_v2[2];
+        frame1Pyr[0].ptr = NULL;
+        frame1Pyr[1].ptr = dst_data;
+        frame1Pyr[1].stride = dst_step;
```

Comment:
I propose to use local variable on stack. new is redundant here. Also it's not deleted afterwards.